src/application.ts

```diff
@@ -83,7 +86,7 @@ export default class Application implements Owner {
     registry.register(`environment:/${this.rootName}/main/main`, Environment);
     registry.registerOption('helper', 'instantiate', false);
     registry.registerOption('template', 'instantiate', false);
-    registry.register(`document:/${this.rootName}/main/main`, window.document as any);
+    registry.register(`document:/${this.rootName}/main/main`, this.document as any);
```

Comment:
@rwjblue This is a general annoyance with the type definitions of `@glimmer/di`'s registry: it assumes everything is a factory, even if it's configured to be a singleton.